casper/src/main/resources/ListOps.rho

```diff
@@ -141,6 +213,45 @@ new  ListOps, rs(`rho:registry:insertSigned:secp256k1`), uriOut in {
     }
   } |
 
+  new mapLoop in {
+    contract mapLoop(@l, @count, function, @state, collect, startCount)= {
+      match l {
+        [hd ...tl] => {
+          function!(state, hd, *collect) | 
+          mapLoop!(tl, count + 1, *function, state, *collect, *startCount)
+        }
+        _  => { startCount!(count) }
+      }
+    } |
+
+    contract ListOps(@"unorderedParMapClean", @list, function, @state, return) = {
+      new collect, startCount, completeCount, accCh in {
+        //Spawn processes, each computing one element
+        //also processes listening for the result
+        //add a result to our accumulating list
+        contract collect(@result) = {
```

Comment:
It looks like `collect` can be extracted in a manner similar to what you've done for the others.

Reply:
No, it can't: `collect` closes over `startCount`, `completeCount`, and `accCh`, but I can't pass those in without changing the signature of `function`.  That kind of refactoring would have to alter the call site.